electrum/wallet.py

```diff
@@ -2360,12 +2360,12 @@ def get_request_status(self, key):
             status = PR_PAID
         return self.check_expired_status(r, status)
 
+    def get_request_by_addr(self, addr: str) -> Optional[Invoice]:
+        key = self._requests_addr_to_key.get(addr)
+        return self._receive_requests.get(key)
+
     def get_request(self, key: str) -> Optional[Invoice]:
-        if req := self._receive_requests.get(key):
-            return req
-        # try 'key' as a fallback address for lightning invoices
-        if (rhash := self._requests_addr_to_rhash.get(key)) and (req := self._receive_requests.get(rhash)):
-            return req
+        return self._receive_requests.get(key)
```

Comment:
note that `wallet.get_request(address)` no longer finds a request now. This is effectively an API change so we need to track down all callers and consider the implications.

For example, consider
- the `delete_request` command
    https://github.com/spesmilo/electrum/blob/21255b33af69a40f42fb9b65870f470af8a213c6/electrum/commands.py#L983-L985
    that currently takes an address (sidenote: we should support deleting by rhash), and would now be broken with this change